--- tests/conftest.py.orig	2025-07-15 12:49:44
+++ tests/conftest.py	2025-07-15 12:51:45
@@ -165,6 +165,13 @@
 
 
 def pytest_sessionstart(session: Session) -> None:  # noqa: U100
+    all_fixtures_ready = all(
+        os.path.exists(props["fixture_path"])
+        for props in TEST_FIXTURES.values()
+    )
+    if os.environ.get("MUTANT_UNDER_TEST") and all_fixtures_ready:
+        return
+
     if get_xdist_worker_id(session) != "master":
         return
 
@@ -197,6 +204,8 @@
     session: Session, exitstatus: int  # noqa: U100
 ) -> None:
     del exitstatus
+    if os.environ.get("MUTANT_UNDER_TEST"):
+        return
     if get_xdist_worker_id(session) != "master":
         return
 
